Ninja compile error for Windows Visual Studio 2022

**Describe the bug**
When attempting to build Mantid using ninja on a Windows OS, with the Visual Studio 2022 installed, I get the following error:

```
$ ninja
[2/22] Building CXX object Framework\DataObjects\CMakeFiles\DataObjects.dir\src\FakeMD.cpp.obj
FAILED: Framework/DataObjects/CMakeFiles/DataObjects.dir/src/FakeMD.cpp.obj
ccache C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx64\x64\cl.exe  /nologo /TP -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_BIND_GLOBAL_PLACEHOLDERS -DBOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DDataObjects_EXPORTS -DENABLE_OPENCASCADE -DGSL_DLL -DH5_BUILT_AS_DYNAMIC_LIB -DHAVE_STDINT_H -DIN_MANTID_DATAOBJECTS -DJSON_DLL -DMS_VISUAL_STUDIO -DNOMINMAX -DPOCO_DLL -DPOCO_ENABLE_CPP11 -DPOCO_NO_AUTOMATIC_LIBS -DPOCO_NO_UNWINDOWS -DWNT -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING -D_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING -D_SILENCE_CXX17_SHARED_PTR_UNIQUE_DEPRECATION_WARNING -D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING -D_USE_MATH_DEFINES -D_WINDOWS -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\DataObjects\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\DataObjects -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\API\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\API -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\HistogramData\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\HistogramData -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Kernel\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Kernel -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Types\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Types -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Json\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Json -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Geometry\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Geometry -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Beamline\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Beamline -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Indexing\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Indexing -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\Parallel\inc -IC:\Users\mlc47243\Documents\mantid-dev\mantid-1\buildninja\Framework\Parallel -external:IC:\Users\mlc47243\AppData\Local\mambaforge\envs\mantid-developer\Library\include -external:IC:\Users\mlc47243\AppData\Local\mambaforge\envs\mantid-developer\Library\include\nexus -external:IC:\Users\mlc47243\AppData\Local\mambaforge\envs\mantid-developer\Library\include\eigen3 -external:W0 /DWIN32 /D_WINDOWS /GR /EHsc   /MP /W3 /bigobj   /experimental:external /external:W0  /Zm160 /Z7 /Ob0 /Od /RTC1 -std:c++17 -MD -openmp /showIncludes /FoFramework\DataObjects\CMakeFiles\DataObjects.dir\src\FakeMD.cpp.obj /FdFramework\DataObjects\CMakeFiles\DataObjects.dir\ /FS -c C:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\DataObjects\src\FakeMD.cpp
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\__msvc_int128.hpp(986): error C2908: explicit specialization; 'std::numeric_limits<T>' has already been instantiated
        with
        [
            T=std::_Unsigned128
        ]
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\__msvc_int128.hpp(986): error C2766: explicit specialization; 'std::numeric_limits<T>' has already been defined
        with
        [
            T=std::_Unsigned128
        ]
C:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\DataObjects\inc\MantidDataObjects\MortonIndex\WideIntImpl.h(87): note: see previous definition of 'numeric_limits<std::_Unsigned128>'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\__msvc_int128.hpp(1372): error C2908: explicit specialization; 'std::numeric_limits<T>' has already been instantiated
        with
        [
            T=std::_Signed128
        ]
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\include\__msvc_int128.hpp(1372): error C2766: explicit specialization; 'std::numeric_limits<T>' has already been defined
        with
        [
            T=std::_Signed128
        ]
C:\Users\mlc47243\Documents\mantid-dev\mantid-1\mantid\Framework\DataObjects\inc\MantidDataObjects\MortonIndex\WideIntImpl.h(87): note: see previous definition of 'numeric_limits<std::_Signed128>'
```

**To Reproduce**
1. Make sure you have `Visual Studio 2022` installed (the same error doesn't happen for 2019)
2. From the source directory, configure your build: `cmake --preset=win-ninja -B=../build`
3. Enter the build folder and attempt to build with `ninja`
4. You should get the error above

**Platform/Version (please complete the following information):**
 - OS: Windows

--- a/Framework/DataObjects/inc/MantidDataObjects/FakeMD.h
+++ b/Framework/DataObjects/inc/MantidDataObjects/FakeMD.h
@@ -10,11 +10,13 @@
 
 #include "MantidAPI/IMDEventWorkspace.h"
 #include "MantidDataObjects/DllConfig.h"
-#include "MantidDataObjects/MDEventWorkspace.h"
+#include "MantidGeometry/IDTypes.h"
 #include "MantidKernel/uniform_int_distribution.h"
 
 namespace Mantid {
 namespace DataObjects {
+
+template <typename MDE, size_t nd> class MDEventWorkspace;
 
 /**
     Provides a helper class to add fake data to an MD workspace

--- a/Framework/DataObjects/inc/MantidDataObjects/MortonIndex/WideInt.h
+++ b/Framework/DataObjects/inc/MantidDataObjects/MortonIndex/WideInt.h
@@ -28,6 +28,7 @@
 #include <iomanip>
 #include <limits>
 #include <ostream>
+#include <random>
 #include <sstream>
 #include <stdexcept>
 #include <stdint.h>

